src/features/PostsAPI/PostsAPI.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+type PostsState = {
+  posts: Post[];
+  loading: boolean;
+  error: string;
```

Comment:
The 'error' property should be of type 'string | null' to properly handle cases where there is no error.